src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -26,27 +26,155 @@ hydrostatic pressure) to the current model state. If `callbacks` are provided (i
 they are called in the end.
 """
 update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[]; compute_tendencies = true) =
-         update_state!(model, model.grid, callbacks; compute_tendencies)
+    update_state!(model, model.grid, callbacks; compute_tendencies)
+
+operation_corner_points = "default" # Choose operation_corner_points to be "default", "average", "CCW", or "CW".
+
+function fill_paired_halo_regions!(fields, signed=true)
+
+    field₁, field₂ = fields
+    grid = field₁.grid
+
+    if !(grid isa ConformalCubedSphereGrid)
+        return
+    end
```

Comment:
`signed` would be cleaner if it was a kwarg.

Also, why not use the `CubedSphereField` type to do something like this:

```suggestion
fill_paired_halo_regions!(fields::Tuple; signed=true) = fill_paired_halo_regions!(fields[1], fields[2]; signed)

function fill_paired_halo_regions!(field₁, field₂; signed=true) = nothing

function fill_paired_halo_regions!(field₁::CubedSphereField, field₂::CubedSphereField; signed=true)

    grid = field₁.grid

```